test/recall.js

```diff
@@ -0,0 +1,22 @@
+var R = require('../source');
+var eq = require('./shared/eq');
+
+var fn = function(a) {
+  var x = a + 1;
+  return function(b) {
+    var y = x * b;
+    return function(c) {
+      return y * c;
+    };
+  };
+};
+
+describe('recall', function() {
+  it('collapses a high-order unary function', function() {
+    eq(R.recall(fn, 1, 2, 3), 12);
+  });
+
+  it('accepts no more than N arguments for an N-order function', function() {
+    eq(R.recall(fn, 1, 2, 3), R.recall(fn, 1, 2, 3, 4));
+  });
+});
```

Comment:
I don't feel that this tests proves what it claims to prove.  Perhaps with an innermost function in `fn` that has an optional parameter and acts differently based on its presence.

```js
var fn = function(a) {
  var x = a + 1;
  return function(b) {
    var y = x * b;
    return function(c, d = 0) {
      return y * c + d;
    };
  };
};
```

where

```js
fn (1) (2) (3)          //=> 12
fn (1) (2) (3, 4)       //=> 16
recall (fn, 1, 2, 3, 4) //=> 12
```

But then that makes me wonder if this is the correct behavior.  It seems to me that if we do this as `:: Int -> (a -> b -> ... -> n -> x) -> ((a, b, ..., n) -> x)`, then we could write this so that 

```js
recall (3, fn) (1, 2, 3, 4) //~> fn (1) (2) (3, 4)
``` 

which feels more in keeping with other Ramda functions.

Reply:
It's intended to support a special class of high-order function, those composed of unary functions. E.g. `const f = g => h => v`, where `f`, `g`, and `h` are unary functions. In `compose`, all functions must be unary except the last because the inputs to the last function are outside the workings of `compose`. Similarily, the last value of the high-order unary function can be anything, including a polyadic function, because it's considered to be the value and outside the workings of `recall`. The tests themselves could use some work though, I agree.

Reply:
Yes, I was responding to the test case.  I don't think it really demonstrated that it "accepts no more than N arguments for an N-order function". But I am curious about which behavior we actually want.  I'm thinking that it might be a little different from what's in this implementation.